### PR TITLE
[Event Hubs] Update to New Repository Structure

### PR DESCRIPTION
**Pull request by jsquire**
# Summary
 
  The Event Hubs area of the repository was structured for compliance with the legacy standards.  With a new standard repository structure having been formally adopted, the Event Hubs client library should conform to it.   
_(see: https://github.com/Azure/azure-sdk/blob/master/docs/engineering-system/repo-structure.md)_

# Known Issues

- The urls in the ReadMe make reference to the new repository location in the `master` branch, which will not exist until this set of changes is merged.  Unfortunately, that makes validation of the links in the ReadMe by simply clicking them not possible.  

# Goals 

- Relocate and restructure the Event Hubs client library to adopt the new standards for repository structure.

- Ensure that references in the ReadMe reflect the new location and structure.

- Organize projects around the intended package units, rather than the implementation.

- Make reasonable efforts to isolate packages such that they are independent of one another; this will not be entirely possible due to shared code and assets, which should be organized under the primary package of the SDK (`Microsoft.Azure.EventHubs`).

# Non-Goals

- Relocation of the management library for Event Hubs.  Due to differences in the engineering system, the management client will continue to exist as part of the legacy structure at present.

- Ensuring stability, isolation, and performance of the Live Test suite; this will be completed as part of another workstream.

# Details

###  Engineering system

- Annotated the document settings to note the known format discrepancies for the current version of the Event Hubs client library ReadMe, as a Track 1 sdk.

- Renamed environment variables to follow the new patterns chosen as the cross-SDK standard.  They are now:  
  
     EVENT_HUBS_CONNECTION_STRING
     EVENT_HUBS_STORAGE_CONNECTION_STRING

###  Project Structure

- Solution and projects have been move to the new sdk root location, with their containing directories renamed to match the standard.

- The directories for shared items (such as Tests) under the client library have been located under the root of the main package (`Microsoft.Azure.EventHubs`) with references to the additional projects.  Though this does impede the goal of having packages entirely independent, it allows for the structure to be organized around packages rather than implementation.

- The ReadMe.md file has been renamed to README.md for consistency with other .NET client 

# Last Upstream Rebase

Thursday, April 25, 2019 9:48am (EDT)

# Related and Follow-Up Issues

- [[Objective][Event Hubs] Migrate Client Library to the Central Repository](https://github.com/Azure/azure-sdk-for-net/issues/5447) (#5447)
- [[Event Hubs] Update to New Repository Structure](https://github.com/Azure/azure-sdk-for-net/issues/5851) (#5851)
- [[Event Hubs Client] Stabilize Live Tests](https://github.com/Azure/azure-sdk-for-net/issues/5451) (#5451)

### CODE REVIEW

File: eng/.docsettings.yml [hunk 1]
```diff
@@ -16,6 +16,8 @@ required_readme_sections:
   - ^Contributing$
 known_presence_issues:
   - ['sdk/keyvault','#5499']
+  - ['sdk/eventhub','#5499']
 known_content_issues:
   - ['README.md','Root readme']
-  - ['sdk/keyvault/Microsoft.Azure.KeyVault/README.md','#5499']
\ No newline at end of file
+  - ['sdk/keyvault/Microsoft.Azure.KeyVault/README.md','#5499']
+  - ['sdk/eventhub/Microsoft.Azure.EventHubs/ReadMe.md','#5499']
```
Comment (serkantkaraca) [inline eng/.docsettings.yml]: >eventhub [](start = 10, length = 8)

Service name is 'EventHubs' actually. We use 'event hub' to refer to single Event Hubs entity.

Comment (jsquire) [reply eng/.docsettings.yml]: Understood, and I've tried to honor that wherever possible.  That said, the repository structure is expected to snap to the names used in the [Azure REST API Specifications](https://github.com/Azure/azure-rest-api-specs/tree/master/specification/eventhub/resource-manager) repository.   Unfortunately, in this case, they used `eventhub` which is why the root namespace lacks the proper `s`.

**Review by serkantkaraca — APPROVED** [review]

File: sdk/eventhub/Directory.Build.props [hunk 2]
```diff
@@ -25,7 +25,7 @@
     Azure messaging client libraries.
   -->
   <PropertyGroup Condition="'$(SignAssembly)' == 'true'">
```
Comment (weshaggard) [inline sdk/eventhub/Directory.Build.props]: We should consider also moving this props file under Microsoft.Azure.EventHubs as well. That way it doesn't get picked up by the track 2 and management libraries when they are moved.  It wouldn't be too bad to move this under Microsoft.Azure.EventHubs and then add a Directory.Build.props file in the other 2 package folders that reference this by relative path.

File: sdk/eventhub/Microsoft.Azure.EventHubs/ReadMe.md [hunk 3]
```diff
@@ -25,7 +25,7 @@ Use the client library for Event Hubs to:
 
 - Receive events from one or more publishers, transform them to better meet the needs of your ecosystem, then publish the transformed events to a new stream for consumers to observe.
```
Comment (weshaggard) [inline sdk/eventhub/Microsoft.Azure.EventHubs/ReadMe.md]: I don't think it is a requirement (at least not yet :)) but for consistency most places name this file README.md in all uppercase. I know that was already the case before the move but given case sensitivity issues on the different OS's the best time to change the case of a file is part of a move like you are doing here.

File: sdk/eventhub/Directory.Build.props (hunk 2 above)
Comment (jsquire) [reply sdk/eventhub/Directory.Build.props]: Good idea... that hadn't occurred to me.   Done!

File: sdk/eventhub/Microsoft.Azure.EventHubs/ReadMe.md (hunk 3 above)
Comment (jsquire) [reply sdk/eventhub/Microsoft.Azure.EventHubs/ReadMe.md]: Done.   _(I'd agree that this is a good time for such things)_

**Comment by jsquire** [comment]
@mikeharder:   With respect to our conversation last night on environment variable patterns across SDKs, I brought these into alignment while I was here:


```
Renamed environment variables to follow the new patterns chosen as the cross-SDK standard. They are now:
    EVENT_HUBS_CONNECTION_STRING
    EVENT_HUBS_STORAGE_CONNECTION_STRING
````

**Review by serkantkaraca — APPROVED** [review]

**Review by AlexGhiondea — APPROVED** [review]

**Review by weshaggard — APPROVED** [review]

**Review by mikeharder — APPROVED** [review]

**Pull request merged**